src/Engine/IntegrityViolation.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Engine\IntegrityViolation.
+ */
+
+namespace Drupal\rules\Engine;
+
+class IntegrityViolation {
+
+  protected $message;
+
+  protected $contextName;
+
+  protected $uuid;
+
+  public function setMessage($message) {
+    $this->message = $message;
```

Comment:
Do we need message translation here? I guess we could require the message to be pre-translated, can we?